### PR TITLE
3.0: Improve delete image command

### PR DESCRIPTION
**Pull request by yuleiwan**
**Please See** [Git Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions)

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/api/pcluster_api.py [hunk 1]
```diff
@@ -388,21 +381,31 @@ def list_images(region: str):
...
-                for imagebuilder in imagebuilder_stacks
+                ImageBuilderStackInfo(imagebuilder=imagebuilder) for imagebuilder in imagebuilder_stacks
             ]
 
             return images_response + imagebuilder_stacks_response
         except Exception as e:
             return ApiFailure(str(e))
+
+    @staticmethod
+    def _get_underlying_image_or_stack(imagebuilder: ImageBuilder, image_id: str):
```
Comment (lukeseawalker) [inline cli/src/pcluster/api/pcluster_api.py:400]: why passing `image_id` as parameter and not retrieve it from `imagebuilder.image_id`?

File: cli/src/pcluster/api/pcluster_api.py [hunk 2]
```diff
@@ -339,14 +338,11 @@ def delete_image(image_id: str, region: str, force: bool = False):
                 os.environ["AWS_DEFAULT_REGION"] = region
             # retrieve imagebuilder config and generate model
             imagebuilder = ImageBuilder(image_id=image_id)
+            image = PclusterApi._get_underlying_image_or_stack(imagebuilder, image_id)[0]
```
Comment (lukeseawalker) [inline cli/src/pcluster/api/pcluster_api.py:341]: isn't better 
`image, _ = PclusterApi._get_underlying_image_or_stack(imagebuilder, image_id)`
? I think it's more readable

File: cli/src/pcluster/api/pcluster_api.py [hunk 3]
```diff
@@ -363,13 +359,10 @@ def describe_image(image_id: str, region: str):
...
 
             imagebuilder = ImageBuilder(image_id=image_id)
-            try:
-                return ImageBuilderImageInfo(imagebuilder=imagebuilder, image=imagebuilder.image)
-            except NonExistingImageError:
-                try:
-                    return ImageBuilderStackInfo(imagebuilder=imagebuilder, stack=imagebuilder.stack)
-                except NonExistingStackError:
-                    raise ImageBuilderActionError(f"Image {image_id} does not exist.")
+            image = PclusterApi._get_underlying_image_or_stack(imagebuilder, image_id)[0]
```
Comment (lukeseawalker) [inline cli/src/pcluster/api/pcluster_api.py:362]: same here

File: cli/src/pcluster/api/pcluster_api.py (hunk 2 above)
Comment (yuleiwan) [reply cli/src/pcluster/api/pcluster_api.py]: done

File: cli/src/pcluster/api/pcluster_api.py (hunk 1 above)
Comment (yuleiwan) [reply cli/src/pcluster/api/pcluster_api.py]: done

**Review by lukeseawalker — APPROVED** [review]

**Pull request merged**